src/formattedtext.jl

```diff
@@ -85,23 +85,88 @@ function Makie.plot!(plot::FormattedText{<:Tuple{<:Markdown.Paragraph}})
+    glyphcollection = Observable{Makie.GlyphCollection}()
+    emojicollection = Observable{Vector{Tuple{String,Int64}}}()
+    onany(text_elements_fonts, plot.textsize, plot.align, plot.rotation, 
+          plot.justification, plot.lineheight, plot.color, plot.strokecolor, plot.strokewidth, 
+          plot.word_wrap_width) do elements_fonts, ts, al, rot, jus, lh, col, scol, swi, www
 
         ts = to_textsize(ts)
         rot = to_rotation(rot)
         col = to_color(col)
         scol = to_color(scol)
 
-        layout_formatted_text(elements_fonts, ts, al, rot, jus, lh, col, scol, swi, word_wrap_width)
+        glc, emc = layout_formatted_text(elements_fonts, ts, al, rot, jus, lh, col, scol, swi, www)
+        glyphcollection[] = glc
+        emojicollection[] = emc
     end
 
-    text!(plot, glyphcollection; plot.attributes...)
-
     notify(plot.text)
+    textplot = text!(plot, glyphcollection; plot.attributes...)
+
+    glyphbbs = lift(glyphcollection) do glc
+        gl_bboxes(glc)
+    end
+
+    textbb = Ref(BBox(0, 1, 0, 1))
+    emoji_positions = Observable{Vector{Point2f}}(Point2f[(0,0)])
+    emoji_images    = Observable{Vector{Matrix{RGBAf}}}(Matrix{RGBAf}[rand(RGBAf, 1,1)])
+    emoji_sizes     = Observable{Vector{Vec2f}}(Vec2f[(0,0)])
+    emoji_offsets   = Observable{Vector{Vec2f}}(Vec2f[(0,0)])
+    onany(emojicollection, plot.align) do ec, align
+        # empty and size all arrays
+        empty!(emoji_positions.val)
+        empty!(emoji_images.val)
+        empty!(emoji_sizes.val)
+        empty!(emoji_offsets.val)
+        sizehint!(emoji_positions.val, length(ec))
+        sizehint!(emoji_images.val, length(ec))
+        sizehint!(emoji_sizes.val, length(ec))
+        sizehint!(emoji_offsets.val, length(ec))
+
+        textbb[] = Rect2f(boundingbox(textplot))
+        box, boy = textbb[].origin
+        w, h = width(textbb[]), height(textbb[])
+        println("(box, boy, w, h) = ($box, $boy, $w, $h)")
+        halign, valign = align
+        anchor_x = if halign === :left
+            box
+        elseif halign === :center
+            box + w/2
+        else #halign === :right
+            box + w
+        end
+        anchor_y = if valign === :top
+            boy + h
+        elseif valign === :center
+            boy + h/2
+        else #valign === :bottom
+            boy
+        end
+        anchor = Point2f(anchor_x, anchor_y)
+
+        positions, char_offsets, quad_offsets, uvs, scales = Makie.text_quads(
+            Vec2f(textplot.attributes.position[][1:2]), glyphcollection[], Vec2f(0), Makie.transform_func_obs(textplot.parent)[])
+
+        for (shorthand, pos) in ec
+            filename = emoji_filename_png(shorthand)
+            push!(emoji_positions.val, Point2f(char_offsets[pos][1:2]) + anchor)
+            push!(emoji_images.val, load_emoji_image(filename))
+            push!(emoji_sizes.val, scales[pos])
+            push!(emoji_offsets.val, quad_offsets[pos])
```

Comment:
`quad_offset` and `scales` include corrections for the padding that's added when converting characters to signed distance fields. You probably don't want that. I think the metrics you need here are `gc.origins`, `minimum(glyph_bb)` and `widths(glyph_bb)`. 

https://github.com/JuliaPlots/Makie.jl/blob/16ff3eb37818cc99cf7ad32857c7d69191914aa9/src/layouting/layouting.jl#L319-L325

For a brief explenation on the padding stuff: Makie encodes characters with signed distance fields. The sign tells you whether you're inside or outside the character, the distance, well, the distance to the closest edge. When rendering a character the distances get interpolated to find the edge and do appropriate antialiasing. For this to work you need a point outside and a point inside the character. To guarantee this is still the case when shrinking the texture, padding needs to be added.

Reply:
I tried with your suggestions (with the bboxes from`FreeTypeAbstraction.metrics_bb` or the e high-insensitive ones returned from`gl_bboxes`)
![image](https://user-images.githubusercontent.com/26469701/173223553-5689dce9-8daf-4c12-8e46-63d98cf0a200.png)

Although not very visible, in the above grid of `FormattedLabels` each label has an `halign, valign` such that it is as close to the center as possible, e.g. the top left label has `(:right, :bottom)` and its horizontal neighbor has `(:center, :bottom)`.

The horizontal offset is still present and the emoji size seems to be off too. At least the latter was correct when using `scales` from `text_quad`.
If I eyeball the horizontal offset, I would say that the offset in the first colon (`:right`) is twice that of the second one (`:center`) and it is ok for the thrid one (`:left`). 
Could this be line wrapping issue? Because when everything fits into one line then the positioning is correct.

Reply:
> The horizontal offset is still present and the emoji size seems to be off too. 

The size is off because the emoji has its own padding. It happens to be the same in this case (12px) but I don't think there is a reason why this would generally be the case.

In regard to offsets - I don't get any with this

```julia
anchor = Point2f(plot.position[])

gc = glyphcollection[]
for (shorthand, pos) in ec
    filename = emoji_filename_png(shorthand)

    glyph_bb, extent = Makie.FreeTypeAbstraction.metrics_bb(
        gc.glyphs[pos], gc.fonts[pos], gc.scales[pos]
    )

    push!(emoji_positions.val, Point2f(gc.origins[pos]) + anchor)
    push!(emoji_images.val, load_emoji_image(filename))
    push!(emoji_sizes.val, widths(glyph_bb))
    push!(emoji_offsets.val, minimum(glyph_bb))
end
```

Reply:
Thought I had tried it that way, but it is now working. Thanks for the snippet.

Reply:
![image](https://user-images.githubusercontent.com/26469701/173237655-42632e50-8060-4e95-a59e-02cbc9bf7674.png)

Any ideas on how to get the size to match?

I tried with 
```julia
            pad = Makie.GLYPH_PADDING[] / Makie.PIXELSIZE_IN_ATLAS[]
            push!(emoji_positions.val, Point2f(gc.origins[pos]) + anchor)
            push!(emoji_images.val, load_emoji_image(filename))
            push!(emoji_sizes.val, widths(glyph_bb) + gc.scales[pos] * 2pad)
            push!(emoji_offsets.val, minimum(glyph_bb))
```
which appears to give the correct size, but then also displaces all emojis towards the top-right corner:
![image](https://user-images.githubusercontent.com/26469701/173237790-703aa899-0b36-4f18-b7a2-3f1dd3f810ba.png)

Reply:
In your case you should have

```julia
EMOJI_PADDING = 12 # if it's constant, otherwise you'd need to know it per pixel and potentially left/right/bottom/top padding
EMOJI_SIZE = 71 # if it's constant, otherwise size(img) probably works
```

and you need to do the same corrections that text_quad has, with those constants

```julia
scaled_pad = EMOJI_PADDING * gc.scales[pos] / EMOJI_SIZE
push!(emoji_sizes.val, widths(glyph_bb) + 2scaled_pad)
push!(emoji_offsets.val, minimum(glyph_bb) - scaled_pad)
```

that should make the placing correct I think

Reply:
Wow, that works perfectly. Thanks.